cmd/basicauth/basicauth.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	b64 "encoding/base64"
+	"flag"
+	"fmt"
+	"net/url"
+)
+
+var (
+	clientId     = flag.String("id", "", "Client ID as string")
+	clientSecret = flag.String("secret", "", "Client secret as string")
+)
+
+func main() {
+	flag.Parse()
+
+	if *clientId == "" || *clientSecret == "" {
+		fmt.Println("Please provide a client ID and secret:")
+		flag.PrintDefaults()
+		return
```

Comment:
exit with non-zero status

```suggestion
		flag.PrintDefaults()
		panic("please provide a client ID and secret")
```